### PR TITLE
Login feature

### PR DESCRIPTION
**Pull request by clalil**
# Login Functionality
[PT-Link](https://www.pivotaltracker.com/story/show/168829437)

## PR content
In this PR we are adding login functionality with the help of the gem Devise.

## This PR contains
Login Views
Feature tests
Happy Path
Sad path
Complementary sad paths to sign up feature

## What I have learned
We have gained a deeper understanding of Devise.

### CODE REVIEW

**Comment by tochman** [comment]
No git ponging... 🤯

**Review by tochman — APPROVED** [review]
`features/sign_up.feature` is out of scope. Please be mindful of that moving forward. Also, I would need you to gitpong more. Other than that, great job.

File: features/support/warden.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Warden.test_mode!
```
Comment (oliverochman) [inline features/support/warden.rb]: why is this added?

Comment (clalil) [reply features/support/warden.rb]: Our bad, we were prematurely adding him.
